straxen/analyses/holoviews_waveform_display.py

```diff
@@ -187,26 +195,63 @@ def _hvdisp_plot_records_2d(records,
     return shader, records, time_stream
 
 
-def plot_record_polygons(record_points, width=1.1, **kwargs):
+def plot_record_polygons(record_points,
+                         center_time=True,
+                         scaling=10**-3,):
     """
     Plots record hv.Points as polygons for record matrix.
 
     :param record_points: Holoviews Points generated with
         _records_to_points.
-    :param width: Length of the record in µs.
-    :param kwargs: Keyword arguments applied to hv.Polygons options.
+    :param center_time: If true treats specified times as center times.
+    :param scaling: Scale times scale by this factor. E.g. if times
+        should be in µs use 10**-3.
     :returns: hv.Polygons
     """
     import holoviews as hv
 
-    data = [{('x', 'y'): rectangle(t, ch, width=width), 'area': a} for t, ch, a in
-            record_points.data[['time', 'channel', 'area']].values]
-    polys = hv.Polygons(data, vdims='area')
-    polys = polys.opts(color='level', aspect=4, responsive='width', line_width=0, **kwargs, cmap='viridis')
+    df = record_points.data.copy()
+    if not center_time:
+        df.loc[:, 'time'] = df.loc[:, 'time'] + (df.loc[:, 'length'] / 2 * df.loc[:, 'dt'])
+
+    data = _make_polys(df,
+                       scaling=scaling)
+    polys = hv.Polygons(data,
```

Comment:
Not very important but im a bit curious about why you chose to use hv.Polygons and not hv.Segments here? isnt the y axis just the channel? so wouldnt all of the polygons be the same height?

Reply:
Nice suggestion, it works like a charm and makes things much easier. Thanks a lot. FYI: the TPC display still uses the old method via hv.Polygons, I will change it when I refactor the TPC event display from bokeh to holoviews.